jujugui/static/gui/src/app/app.js

```diff
@@ -2585,6 +2591,18 @@ YUI.add('juju-gui', function(Y) {
           );
         }, this);
       }
+      // Retrieve maas credentials should they exist.
+      // NB: this assumes that a maas cloud cannot be added to a multi-cloud
+      // controller. If this changes in the future, this code will need to be
+      // called somewhere else. - Makyo 2017-02-16
+      this.controllerAPI.getDefaultCloudName((error, name) => {
+        this.controllerAPI.getClouds([name], (error, data) => {
```

Comment:
Before doing the second call to getClouds we need to check if there is an error returned by the first call (getDefaultCloudName), in which case I'd just stop the execution and log the error in the console.